input-profile.ts

```diff
@@ -0,0 +1,280 @@
+                    ? this.customProfileFolder
+                    : this.path + xrInputSource.profiles[0];
+            this.url = profile + '/profile.json';
+
+            this._profileJSON = InputProfile.Cache.has(this.url)
+                ? InputProfile.Cache.get(this.url)
+                : null;
+
+            if (this._profileJSON != null) return;
+            fetch(this.url)
+                .then((res) => res.json())
+                .then((out) => {
+                    this._profileJSON = out;
+                    console.log('Profile downloaded and loaded from ' + this.url);
+                    InputProfile.Cache.set(this.url, this._profileJSON);
+                    this.loadAndMapGamepad(profile, xrInputSource);
+                })
+                .catch((e) => {
+                    console.error('failed to load profile from' + this.url);
+                    console.error(e);
+                });
+        });
+    }
+
+    isModelLoaded() {
+        return this._controllerModel !== null;
+    }
+
+    async loadAndMapGamepad(profile: string, xrInputSource: XRInputSource) {
+        this._gamepad = xrInputSource.gamepad;
+        const assetPath = profile + '/' + this._handedness + '.glb';
+        this._controllerModel = this.defaultController;
+        if (!this.mapToDefaultController) {
+            /** load 3d model in the runtime with profile url */
+            try {
+                this._controllerModel = (await this.engine.scene.append(
+                    assetPath
+                )) as Object3D;
+            } catch (e) {
+                console.error('failed to load 3d model');
+                console.error(e);
+                this._setComponentsActive(true);
+                console.log(
+                    'Couldnot load i-p, continuing with ' +
+                        this._handedness +
+                        ' default controller'
+                );
+            }
+            this._setComponentsActive(false);
+            this._controllerModel.parent = this.object;
+            this._controllerModel.setPositionLocal([0, 0, 0]);
+            console.log('Disabling ' + this._handedness + ' default Controller');
+            console.log(this._handedness + 'controller model loaded to the scene');
+            if (this.addVrModeSwitch)
+                this._controllerModel.addComponent(VrModeActiveSwitch);
+            this.notifyUrlReady();
+        } else {
+            console.log('mapping i-p to ' + this._handedness + ' default controllers');
+        }
+        this.getGamepadObjectsFromProfile(this._profileJSON, this._controllerModel);
+        this.setHandTrackingControllers(this.defaultController);
+        this.update = () => this.mapGamepadInput();
+    }
+
+    getGamepadObjectsFromProfile(profile: any, obj: Object3D) {
+        const components = profile.layouts[this._handedness].components;
+        if (!components) return;
+
+        for (const i in components) {
```

Comment:
Is it possible to override the profile with another one? If yes you need to clear the `_buttons` and `_axes` array here, otherwise you might end up with extra values